libs/commons/Hashtbl_.ml

```diff
@@ -70,3 +70,13 @@ let peek_opt (tbl : ('k, 'v list ref) Hashtbl.t) (key : 'k) : 'v option =
 let get_stack tbl key =
   try !(Hashtbl.find tbl key) with
   | Not_found -> []
+
+let find_some_opt tbl (f : 'v -> 'a option) key =
+  match Hashtbl.find_opt tbl key with
+  | Some v -> f v
+  | None -> None
```

Comment:
Similar this is just `Hashtbl.find_opt tbl key >>= f`.

Reply:
:( ok